gef.py

```diff
@@ -5736,234 +5729,11 @@ def get_fd_from_result(self, res: str) -> int:
-            return
-
-        ida_added, ida_removed = rc
-
-        # add new bp from IDA
-        for new_bp in ida_added:
-            location = base_address + new_bp
-            gdb.Breakpoint(f"*{location:#x}", type=gdb.BP_BREAKPOINT)
-            self.old_bps.add(location)
-
-        # and remove the old ones
-        breakpoints = gdb.breakpoints() or []
-        for bp in breakpoints:
-            if bp.enabled and not bp.temporary:
-                if bp.location[0] == "*": # if it's an address i.e. location starts with "*"
-                    addr = parse_address(bp.location[1:])
-                else:  # it is a symbol
-                    addr = int(gdb.parse_and_eval(bp.location).address)
-
-                if not (base_address <= addr < end_address):
-                    continue
-
-                if (addr - base_address) in ida_removed:
-                    if (addr - base_address) in self.old_bps:
-                        self.old_bps.remove((addr - base_address))
-                    bp.delete()
-        return
-
-    def usage(self, meth: Optional[str] = None) -> None:
-        if self.sock is None:
-            return
-
-        if meth is not None:
-            gef_print(titlify(meth))
-            gef_print(self.sock.system.methodHelp(meth))
-            return
-
-        info("Listing available methods and syntax examples: ")
-        for m in self.sock.system.listMethods():
-            if m.startswith("system."): continue
-            gef_print(titlify(m))
-            gef_print(self.sock.system.methodHelp(m))
-        return
-
-    def import_structures(self, structs: Dict[str, List[Tuple[int, str, int]]]) -> None:
-        if self.version[0] != "IDA Pro":
-            return
-
-        path = gef.config["pcustom.struct_path"]
-        if path is None:
-            return
-
-        if not os.path.isdir(path):
-            gef_makedirs(path)
-
-        for struct_name in structs:
-            fullpath = pathlib.Path(path) / f"{struct_name}.py"
-            with fullpath.open("w") as f:
-                f.write("from ctypes import *\n\n")
-                f.write("class ")
-                f.write(struct_name)
-                f.write("(Structure):\n")
-                f.write("    _fields_ = [\n")
-                for _, name, size in structs[struct_name]:
-                    name = bytes(name, encoding="utf-8")
-                    if size == 1: csize = "c_uint8"
-                    elif size == 2: csize = "c_uint16"
-                    elif size == 4: csize = "c_uint32"
-                    elif size == 8: csize = "c_uint64"
-                    else:           csize = f"c_byte * {size}"
-                    m = f'        (\"{name}\", {csize}),\n'
-                    f.write(m)
-                f.write("]\n")
-        ok(f"Success, {len(structs):d} structure{'s' if len(structs) > 1 else ''} imported")
+        err("`ida-interact` was removed from gef. You can find a better version as part of `gef-extras`")
```

Comment:
Should we put a ticket or issue to remind us to remove completely after a few releases? Maybe give a deprecation warning here?

Reply:
that works. I just want us to be able to remove it completely :)